baybe/objectives/desirability.py

```diff
@@ -147,7 +144,7 @@ def transform(self, data: pd.DataFrame) -> pd.DataFrame:  # noqa: D102
             transformed[target.name] = target.transform(data[[target.name]])
 
         # Scalarize the transformed targets into desirability values
-        vals = scalarize(transformed.values, self.scalarizer, self.weights)
+        vals = scalarize(transformed.values, self.scalarizer, self._normalized_weights)
```

Comment:
You need to change the comment, since this is also the point where normalization happens, which is not obvious imo.

Reply:
But I don't see what's wrong with the comment? The step is literally the transformation into desirability values. Explicitly mentioning something like "normalize weights and ..." would even be bad because we are not calling a function but a property (which is to be interpreted like an attribute), if that is what you mean. So this should be also handled text-wise as if it was an attribute, because the computation is hidden away at this point.

Reply:
True, I was interpreting the `_normalized_weights` here as a function, which it isn't.